### PR TITLE
use lora_plan library

### PR DESCRIPTION
**Pull request by mikev**

### CODE REVIEW

**Review by ruttenberg — APPROVED** [review]

**Comment by michaeldjeffrey** [comment]
Can we swap out the implementation of the functions in the `pp_lorawan` module, and remove the resulting dead code?

Also, is packet-purchaser still on the latest for the lib?

File: src/pp_roaming_protocol.erl [hunk 1]
```diff
@@ -362,7 +371,8 @@ handle_xmitdata_req(#{
         {error, _} = Err ->
             Err;
         {ok, TransactionID, Region, PacketTime, DestURL, FlowType} ->
-            DataRate = pp_lorawan:index_to_datarate(Region, DR),
+            Plan = lora_plan:region_to_plan(Region),
+            DataRate = lora_plan:datarate_to_string(Plan, DR),
```
Comment (marcsugiyama) [inline src/pp_roaming_protocol.erl:375]: This pair looks like an idiom. Maybe create a utility function to convert a Region to DataRate and hide the Plan?

File: src/pp_roaming_protocol.erl [hunk 2]
```diff
@@ -198,13 +204,14 @@ handle_prstart_ans(#{
...
     {ok, TransactionID, Region, PacketTime, _, _} = parse_uplink_token(Token),
-
+    Plan = lora_plan:region_to_plan(Region),
+    DataRate = lora_plan:datarate_to_string(Plan, DR),
     DownlinkPacket = blockchain_helium_packet_v1:new_downlink(
         pp_utils:hexstring_to_binary(Payload),
         _SignalStrength = 27,
         pp_utils:uint32(PacketTime + ?JOIN1_DELAY),
         Frequency,
-        pp_lorawan:index_to_datarate(Region, DR),
```
Comment (marcsugiyama) [inline src/pp_roaming_protocol.erl:207]: Is `pp_lorawan:index_to_datarate/2` used anywhere? If not, should we remove it?

**Review by marcsugiyama — APPROVED** [review]

File: src/pp_roaming_protocol.erl (hunk 1 above)
Comment (michaeldjeffrey) [reply src/pp_roaming_protocol.erl:375]: This is what the `pp_lorawan` module is for.

File: src/pp_roaming_protocol.erl (hunk 2 above)
Comment (mikev) [reply src/pp_roaming_protocol.erl:207]: That was the idea :-)  Done.

**Pull request closed without merging**